integration_tests/src/tests/evm_notification_test.rs

```diff
@@ -0,0 +1,381 @@
+    );
+
+    // Deploy the NotificationReceiver EVM contract
+    let hex_str = std::fs::read_to_string("../actors/evm/tests/contracts/NotificationReceiver.hex")
+        .expect("Failed to read contract bytecode hex file");
+    let hex_str = hex_str.trim();
+    let contract_bytecode = hex::decode(hex_str).expect("Failed to decode contract bytecode hex");
+
+    // Create an EVM actor to receive notifications
+    let params =
+        IpldBlock::serialize_cbor(&fil_actor_eam::CreateExternalParams(contract_bytecode)).unwrap();
+
+    let create_result = v
+        .execute_message(
+            &worker,
+            &EAM_ACTOR_ADDR,
+            &TokenAmount::zero(),
+            fil_actor_eam::Method::CreateExternal as u64,
+            params,
+        )
+        .unwrap();
+
+    assert!(
+        create_result.code.is_success(),
+        "Failed to create EVM contract: {}",
+        create_result.message
+    );
+
+    let create_return: fil_actor_eam::CreateReturn =
+        create_result.ret.unwrap().deserialize().expect("Failed to decode create return");
+    let evm_robust_addr = create_return.robust_address.unwrap();
+    let _evm_eth_addr = create_return.eth_address;
+
+    // Precommit sectors
+    let sector_number: SectorNumber = 100;
+
+    // Create piece activation manifests with notifications to EVM contract
+    let piece_size0 = PaddedPieceSize(32 << 30); // 32 GiB
+    let piece_cid0 = make_piece_cid(format!("piece-{}", 0).as_bytes());
+    let notification_payload = RawBytes::from(hex::decode("cafe").unwrap());
+
+    let manifests: Vec<SectorActivationManifest> = vec![SectorActivationManifest {
+        sector_number,
+        pieces: vec![PieceActivationManifest {
+            cid: piece_cid0,
+            size: piece_size0,
+            verified_allocation_key: None,
+            notify: vec![
+                // Send notification to our EVM contract
+                DataActivationNotification {
+                    address: evm_robust_addr,
+                    payload: notification_payload.clone(),
+                },
+            ],
+        }],
+    }];
+
+    let meta: Vec<PrecommitMetadata> = manifests
+        .iter()
+        .map(|sector| {
+            let pis: Vec<PieceInfo> =
+                sector.pieces.iter().map(|p| PieceInfo { size: p.size, cid: p.cid }).collect();
+            let commd = v.primitives().compute_unsealed_sector_cid(seal_proof, &pis).unwrap();
+            PrecommitMetadata { deals: vec![], commd: CompactCommD::of(commd) }
+        })
+        .collect();
+
+    // Track the precommit epoch for later verification
+    let precommit_epoch = v.epoch();
+
+    precommit_sectors_v2(v, 1, meta, &worker, &miner_addr, seal_proof, sector_number, true, None);
+
+    // Before prove commit no notifications have been received
+    check_receiver_notification_count(v, &worker, &evm_robust_addr, 0);
+
+    // Advance time to prove commit epoch
+    let prove_time = v.epoch() + 151;
```

Comment:
```suggestion
    let prove_time = v.epoch() + PRE_COMMIT_CHALLENGE_DELAY + 1;
```

not important, take it or leave it, just a suggestion to make it clear what we're doing here